postgis/lwgeom_box3d.c

```diff
@@ -601,3 +621,384 @@ Datum BOX3D_construct(PG_FUNCTION_ARGS)
 
 	PG_RETURN_POINTER(result);
 }
+
+/*****************************************************************************
+ * BOX3D functions
+ *****************************************************************************/
+
+/* contains? */
+
+bool
+BOX3D_contains_internal(BOX3D *box1, BOX3D *box2)
+{
+	return (FPge(box1->xmax, box2->xmax) &&
```

Comment:
I'm not quite sure FPge is a fit here. A box that is unbuffered by 9E-7 will be containing the original one? It's also strange that there's 1e-8 SQLMM epsilon, and FPge uses a 100 times bigger one.

```
#define EPSILON					1.0E-06
#define FPge(A,B)				((B) - (A) <= EPSILON)
```

Reply:
Indeed, we could use float8_cmp_internal instead of those FP**

Reply:
can it be just `>=` ?

Reply:
The difference with float8_cmp_internal is that it takes care of NaN. Please tell me which is better.